java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java

```diff
@@ -47,23 +50,36 @@ public void start(Map<String, String> props) {
         context -> ofNullable(context.offsetStorageReader())).orElseThrow();
     AzureEventHubsConfig azureEventHubsConfig = new AzureEventHubsConfig(props);
     TopicPartitionOffsetProvider topicPartitionOffsetProvider = new TopicPartitionOffsetProvider(offsetStorageReader);
+    Kcql parsedKcql = Kcql.parse(
+        azureEventHubsConfig.getString(AzureEventHubsConfigConstants.KCQL_CONFIG).split(";")[0]);
+    KeyValueTypes keyValueTypes = getSourceDataTypes(parsedKcql);
```

Comment:
You mentioned that this parsing and splitting thing is duplicated in some of the scala code, this would be a very simple thing to push to the KCQL lib so as to avoid the duplication.